File: TopShelf/PVRecentGame+TopShelf.m
//
//  PVRecentGame+TopShelf.m
//  Provenance
//
//  Created by David Muzi on 2015-12-15.
//  Copyright © 2015 James Addyman. All rights reserved.
//

#import "PVRecentGame+TopShelf.h"
#import "PVEmulatorConstants.h"
#import "PVAppConstants.h"

@implementation PVRecentGame (TopShelf)

- (TVContentItem *)contentItemWithIdentifier:(TVContentIdentifier *)containerIdentifier
{
    TVContentIdentifier *identifier = [[TVContentIdentifier alloc] initWithIdentifier:self.game.md5Hash container:containerIdentifier];
    TVContentItem *item = [[TVContentItem alloc] initWithContentIdentifier:identifier];
    
    item.title = self.game.title;
    item.imageURL = [NSURL URLWithString:self.game.originalArtworkURL];
    item.imageShape = self.imageType;
    item.displayURL = self.displayURL;
    item.lastAccessedDate = self.lastPlayedDate;
    
    return item;
}

- (NSURL *)displayURL
{
    NSURLComponents *components = [[NSURLComponents alloc] init];
    components.scheme = PVAppURLKey;
    components.path = PVGameControllerKey;
    components.queryItems = @[[[NSURLQueryItem alloc] initWithName:PVGameMD5Key value:self.game.md5Hash]];
    
    return components.URL;
}

- (TVContentItemImageShape)imageType
{
    if ([self.game.systemIdentifier isEqualToString:PVNESSystemIdentifier] ||
        [self.game.systemIdentifier isEqualToString:PVGenesisSystemIdentifier] ||
        [self.game.systemIdentifier isEqualToString:PVMasterSystemSystemIdentifier]) {
        return TVContentItemImageShapePoster;
    }
    else if ([self.game.systemIdentifier isEqualToString:PVGameGearSystemIdentifier]) {
        return TVContentItemImageShapeSquare;
    }

    return TVContentItemImageShapeHDTV;
}

@end

File: TopShelf/PVRecentGame+TopShelf.swift
//  Converted to Swift 4 by Swiftify v4.1.6613 - https://objectivec2swift.com/
//
//  PVRecentGame+TopShelf.m
//  Provenance
//
//  Created by David Muzi on 2015-12-15.
//  Copyright © 2015 James Addyman. All rights reserved.
//

import PVLibrary
import PVSupport
import TVServices

// Top shelf extensions
extension PVRecentGame {
    public func contentItem(with containerIdentifier: TVContentIdentifier) -> TVContentItem? {
        guard let game = game else {
            return nil
        }

        let identifier = TVContentIdentifier(identifier: game.md5Hash, container: containerIdentifier)
        let item = TVContentItem(contentIdentifier: identifier)

        item.title = game.title
        item.imageURL = URL(string: game.customArtworkURL.isEmpty ? game.originalArtworkURL : game.customArtworkURL)
        item.imageShape = game.system.imageType
        item.displayURL = displayURL
        item.lastAccessedDate = lastPlayedDate

        return item
    }

    var displayURL: URL {
        guard let game = game else {
            ELOG("Nil game")
            return URL(fileURLWithPath: "")
        }

        var components = URLComponents()
        components.scheme = PVAppURLKey
        components.path = PVGameControllerKey
        components.queryItems = [URLQueryItem(name: PVGameMD5Key, value: game.md5Hash)]
        return components.url ?? URL(fileURLWithPath: "")
    }
}
